3sum.java

```diff
@@ -0,0 +1,40 @@
+// space complexity : O(1)
+// time complexity : O(n2)
+class Solution {
+    public List<List<Integer>> threeSum(int[] nums) {
+        List<List<Integer>> list=new ArrayList<>();
+        if(nums.length<3 || nums==null) return list;
```

Comment:
nums == null will be first condition as if nums is null  then nums.length will fail